frame/fast-unstake/src/lib.rs

```diff
@@ -214,6 +216,8 @@ pub mod pallet {
 		/// the chain's resources.
 		#[pallet::weight(<T as Config>::WeightInfo::register_fast_unstake())]
 		pub fn register_fast_unstake(origin: OriginFor<T>) -> DispatchResult {
+			ensure!(ErasToCheckPerBlock::<T>::get() != 0, <Error<T>>::CallNotAllowed);
```

Comment:
should be after origin check -- origin check does not need storage read and MUST always be the first check. Otherwise this is DOS possibility: someone sending continuous bar transactions with e.g. origin `None` and causing useless storage reads on nodes.

Reply:
Yes, so I had this idea too, but `just a ensure!(ErasToCheckPerBlock != 0, CallNotAllowed) at the beginning of both calls` confused me.